lib/ssl/src/ssl_config.erl

```diff
@@ -2179,27 +2174,41 @@ do_handle_cert_file(File, PemCacheName) ->
 handle_key_file(#{keyfile := File} = CertKey, PemCacheName) ->
     case file:read_file(File) of
         {ok, Pem} ->
-            case public_key:pem_decode(Pem) of
+            PemEntries = public_key:pem_decode(Pem),
+            Password = maps:get(password, CertKey, ""),
+            case key_entry(PemEntries) of
                 [KeyEntry] ->
-                    Password = maps:get(password, CertKey, ""),
                     try public_key:pem_entry_decode(KeyEntry, Password) of
                         Key ->
-                            handle_key(PemCacheName, File, Key, [KeyEntry])
+                            handle_key(PemCacheName, File, Key, PemEntries, Password)
                     catch _:_ ->
                             {error, wrong_password}
                     end;
-                Unexpected ->
-                    {error, {unexpected_content, Unexpected}}
+                _ ->
+                    {error, {unexpected_content, {missing_single_key, File}}}
             end;
         {error, _} =  Error ->
             Error
     end;
 handle_key_file(_,_) ->
     ok.
 
-handle_key(PemCacheName, File, Key, Content) ->
+key_entry(PemEntries) ->
+    [PemEntry || PemEntry = {PKey, _ , _} <- PemEntries,
+                 PKey =:= 'RSAPrivateKey' orelse
+                     PKey =:= 'DSAPrivateKey' orelse
+                     PKey =:= 'ECPrivateKey' orelse
+                     PKey =:= 'PrivateKeyInfo'].
+
+handle_key(PemCacheName, File, Key, PemEntries, Password) ->
     case check_key(Key) of
         ok ->
+            Content = case PemEntries of
+                          [_] = KeyEntry ->
+                              KeyEntry;
+                          [_|_] ->
+                              [public_key:pem_entry_decode(Entry, Password) || Entry <- PemEntries]
+                      end,
```

Comment:
this will lead to `{error, {options, {certfile, ..., no_certs}}}` in `ssl:handshake` because https://github.com/erlang/otp/blob/34316363201de07437e14e79ef559fde7becb0ee/lib/ssl/src/ssl_certificate.erl#L188-L192
does not handle decoded certificates

Reply:
Thanks for the quick feedback, I had not written a new test case yet and only run a ssl_api_SUITE, so maybe I should waited  to push the update. Pushed a  better version.